newrelic/resource_newrelic_cloud_gcp_link_account.go

```diff
@@ -0,0 +1,148 @@
+package newrelic
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/cloud"
+)
+
+func resourceNewRelicCloudGcpLinkAccount() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNewRelicCloudGcpLinkAccountCreate,
+		ReadContext:   resourceNewRelicCloudGcpLinkAccountRead,
+		UpdateContext: resourceNewRelicCloudGcpLinkAccountUpdate,
+		DeleteContext: resourceNewRelicCloudGcpLinkAccountDelete,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "name of the linked account",
+				Required:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "project id of the Gcp account",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceNewRelicCloudGcpLinkAccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	accountID := selectAccountID(providerConfig, d)
+	linkAccountInput := expandGcpCloudLinkAccountInput(d)
+	cloudLinkAccountPayload, err := client.Cloud.CloudLinkAccountWithContext(ctx, accountID, linkAccountInput)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	var diags diag.Diagnostics
+	if len(cloudLinkAccountPayload.Errors) > 0 {
+		for _, err := range cloudLinkAccountPayload.Errors {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  err.Type + " " + err.Message,
+			})
+		}
+		return diags
+	}
+	d.SetId(strconv.Itoa(cloudLinkAccountPayload.LinkedAccounts[0].ID))
```

Comment:
Should we do a quick length check here? If cloudLinkAccountPayload returns an empty array this will crash the plugin.